### PR TITLE
sdk/state: implement CAP-40 on open

### PR DESCRIPTION
**Pull request by leighmcculloch**
For #179

### CODE REVIEW

File: sdk/state/open.go [hunk 1]
```diff
@@ -90,14 +103,40 @@ func (c *Channel) openTxs(d OpenAgreementDetails) (decl, close, formation *txnbu
...
+	declTx, closeTx, formationTx, err := c.openTxs(openAgreement.Details)
 	if err != nil {
-		return nil, fmt.Errorf("building declaration and close txs for latest close agreement: %w", err)
+		return nil, fmt.Errorf("building txs for for open agreement: %w", err)
 	}
 	formationTx, err = formationTx.AddSignatureDecorated(openAgreement.FormationSignatures...)
 	if err != nil {
 		return nil, fmt.Errorf("attaching signatures to formation tx for latest close agreement: %w", err)
 	}
+	for _, s := range openAgreement.DeclarationSignatures {
```
Comment (acharb) [inline sdk/state/open.go:114]: 💡 for these steps of finding the specific signature I think we should add comments. w/o context it's not completely clear why we have to do this imo

File: sdk/txbuild/formation.go [hunk 2]
```diff
@@ -7,26 +7,50 @@ import (
...
+	StartSequence     int64
+	Asset             txnbuild.Asset
+	ExpiresAt         time.Time
+	DeclarationTxHash [32]byte
+	CloseTxHash       [32]byte
+	ConfirmingSigner  *keypair.FromAddress
 }
 
 func Formation(p FormationParams) (*txnbuild.Transaction, error) {
+	extraSignerKeys := [2]xdr.SignerKey{}
```
Comment (acharb) [inline sdk/txbuild/formation.go:27]: 💭 similarly a comment here

**Review by acharb — APPROVED** [review]
overall makes sense to me 👍 . It was a bit hard for me to review because we aren't adding tests in this PR, but glad those are coming in https://github.com/stellar/experimental-payment-channels/issues/193

added a couple spots I think could be good candidates for comments as well

File: sdk/txbuild/formation.go (hunk 2 above)
Comment (leighmcculloch) [reply sdk/txbuild/formation.go:34]: Good point. c4ae700

File: sdk/state/open.go (hunk 1 above)
Comment (leighmcculloch) [reply sdk/state/open.go:117]: Good point. This code was gnarly enough to author let alone read. 07d1f8c

**Pull request merged**